### PR TITLE
Update Node.js GAX package generation

### PR DESCRIPTION
**Pull request by jmuk**
- Modify the problems around generation of index.js files
- Fix package.json contents
- Some other changes

updates #97

### CODE REVIEW

**Comment by jmuk** [comment]
@swcloud, please take a look. I think I can ignore the coverage report failures -- 91.62% is still good enough.

File: config/dependencies.yml [hunk 1]
```diff
@@ -75,4 +75,7 @@ googleapis_common_protos:
 nodeModules:
   arguejs:
     version: '0.2.3'
-
+  extend:
+    version: '3.0.0'
+  lodash:
+    version: '4.6.0'
```
Comment (swcloud) [inline config/dependencies.yml]: I think we need a dependency like

google-proto-files:
       version:  '0.7.0'

File: templates/gax/nodejs/package.json.mustache [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+    "src"
+  ],
   "dependencies": {
     "arguejs": "^{{{dependencies.nodeModules.arguejs.version}}}",
     "google-gax": "^{{{dependencies.gax.nodejs.version}}}",
-    "grpc": "^{{{dependencies.grpc.nodejs.version}}}",
-    "{{{api.dependsOn}}}-{{{api.version}}}": "^{{{api.semantic_version}}}"
+    "extend": "^{{{dependencies.nodeModules.extend.version}}}",{{^singleFile}}
+    "lodash.union": "^{{{dependencies.nodeModules.lodash.version}}}",{{/singleFile}}
   },
```
Comment (swcloud) [inline templates/gax/nodejs/package.json.mustache]: I think we need a dependency

"google-proto-files": "^0.7.0"

File: test/fixtures/gax/nodejs/package.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  "description": "Packager Unittest library for Node.js",
+  "main": "src/index.js",
+  "files": [
+    "src"
+  ],
+  "dependencies": {
+    "arguejs": "^0.2.3",
+    "google-gax": "^0.7.0",
+    "extend": "^3.0.0",
+    "lodash.union": "^4.6.0",
```
Comment (swcloud) [inline test/fixtures/gax/nodejs/package.json]: I think we need a dependency

"google-proto-files": "^0.7.0"

File: test/packager.js [hunk 4]
```diff
@@ -178,6 +185,17 @@ var testPackageInfo = {
...
+    nodeModules: {
+      arguejs: {
+        version: '0.2.3'
+      },
+      extend: {
+        version: '3.0.0'
+      },
+      lodash: {
+        version: '4.6.0'
+      }
```
Comment (swcloud) [inline test/packager.js]: I think we need a dependency

"google-proto-files": "^0.7.0"

File: config/dependencies.yml (hunk 1 above)
Comment (jmuk) [inline config/dependencies.yml]: I think it is in a quite similar position as 'googleapis_common_protos' section, so I added there.

File: templates/gax/nodejs/package.json.mustache (hunk 2 above)
Comment (jmuk) [inline templates/gax/nodejs/package.json.mustache]: Indeed. Done.

File: test/packager.js (hunk 4 above)
Comment (jmuk) [inline test/packager.js]: Done (to googleapis_common_protos)

**Comment by swcloud** [comment]
LGTM

**Pull request merged**